src/translator/translator.js

```diff
@@ -64,27 +65,64 @@ class GoogleTranslator {
 
       return response.data.data.translations[0];
     } catch (e) {
-      if (e.response) {
-        const err = new GoogleApiError(e.response);
-        logger.error(
-          `Google Translation API Error: ${JSON.stringify(
-            err.details,
-            null,
-            2
-          )}`
-        );
-        logger.error(err);
-        throw err;
-      } else {
-        logger.error(`Google Translation API Error: ${e.message}`);
-        throw e;
-      }
+      this._handleAxiosError(e);
     }
   }
 
   _getISOLanguageCode(language) {
     return ISO6391.getCode(language) || language; // if `language` was already an ISO-639-1 code, getCode() will return '' and return value will be defaulted to `language` itself
   }
+
+  _getISOLanguageName(code) {
+    return {
+      name: ISO6391.getName(code) || '',
+      native: ISO6391.getNativeName(code) || '',
+    }; // if `language` was already an ISO-639-1 code, getCode() will return '' and return value will be defaulted to `language` itself
+  }
+
+  async _getSupportedLanguages() {
+    try {
+      const response = await axios.post(
+        GOOGLE_TRANSLATION_API_BASE_URL + '/languages',
+        {},
+        {
+          params: {
+            key: this.apiKey,
+          },
+        }
+      );
+
+      const languages = response.data.data.languages.map(data => {
+        const fullData = this._getISOLanguageName(data.language);
+        fullData.code = data.language;
+        return fullData;
+      });
+
+      this._cachedSupportedLanguages = languages;
+
+      return this._cachedSupportedLanguages;
+    } catch (e) {
+      this._handleAxiosError(e);
+    }
+  }
+
+  _getCachedSupportedLanguages() {
```

Comment:
similar to above, you can remove the `_` from here since it's being used by outsiders